src/main/java/managers/PageObjectManager.java

```diff
@@ -1,18 +1,34 @@
 package managers;
 
+import base.BasePage;
+import org.junit.Assert;
 import org.openqa.selenium.WebDriver;
-import pages.*;
+import pages.HomePage;
+
+import java.lang.reflect.Method;
 
 public class PageObjectManager {
     private WebDriver driver;
     private HomePage homePage;
+
     public PageObjectManager(WebDriver driver) {
         this.driver = driver;
     }
 
-    public HomePage getHomePage () {
-      return (homePage == null) ? homePage = new HomePage(this.driver) : homePage;
+    public HomePage getHomePage() {
+        return (homePage == null) ? homePage = new HomePage(this.driver) : homePage;
     }
 
+    public BasePage getPageByName(String pageName) {
+        pageName = pageName.replace(" ", "");
 
+        try {
+            Method method = PageObjectManager.class.getMethod("get" + pageName + "Page");
+            return (BasePage) method.invoke(this);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
maybe logger instead of printstacktrace?

Reply:
totally agree